python_dependency_verifier/python_dependency_verifier.py

```diff
@@ -137,63 +53,150 @@ def get_latest_version_number(package_name):
+            file_contents = file_contents[:ends_at]
+            self._list_of_unprocessed_dependencies += file_contents.split(",")
+        return self._list_of_unprocessed_dependencies
+
+    def _get_dependency_with_latest_version(self, list_of_dependencies):
+        """receives a list of items from install_requires field
+        in the form of ITEMNAME RELATION VERSION
+        like pika==0.9.13
+        and returns the following dictionary:
+        [name, is_locked_to_specific_version, version_locked_to,
+        version available from pip]
+        per item"""
+        return [self.string_dependency_to_dict(dependency)
+                for dependency in list_of_dependencies if dependency]
+
+    def _process_dependency_list(self):
+        self._list_of_unprocessed_dependencies = remove_quotes_and_whitespace_from_list(
+            self._list_of_unprocessed_dependencies)
+        self._list_of_unprocessed_dependencies = filter_list_for_regex(
+            self._list_of_unprocessed_dependencies, self._ignore_this)
+        self._list_of_dependencies = \
+            [string_dependency_to_dict(dependency)
+             for dependency in self._list_of_unprocessed_dependencies]
+
+    def check_file(self):
+        file_contents = get_file_contents(self._path)
+        self._get_append_list(file_contents)
+        self._get_install_requires_field_contents(file_contents)
+        self._process_dependency_list()
+
+    def get_list_of_dependencies(self):
+        return self._list_of_dependencies
+
+
+class PythonSetuptoolsDependencyCheckerForDir():
+
+    def __init__(self, path_to_recurse, regex_to_ignore):
+        self._path = path_to_recurse
+        self._ignore_this = regex_to_ignore
+        self._result = []
+
+    def check_all_filename_in_subdirs(self):
+        """receives a path and setup filename to search for,
+        a field dependency name and regex to ignore
+        returns a dict of filename and the list of dependencies it uses,
+        whether they are version locked,
+        which version and the latest version of the package available on pip
+        """
+        for root, _, files in os.walk(self._path):
+            for f in files:
+                if f == "setup.py":
+                    fullpath = os.path.join(root, f)
+                    logging.debug("checking file: {0}".format(fullpath))
+                    check_file = PythonSetuptoolsDependencyCheckerForFile(fullpath,self._ignore_this)
+                    check_file.check_file()
+                self._result.append({"filename": fullpath,
+                                         "analysis": check_file.get_list_of_dependencies()})
+        return self._result
+
 
 # path = '/Users/gilzellner/dev/git/cloudify-cosmo/'
-# filename = "/setup.py"
-# field_dependency_name = "install_requires=["
-# check_all_filename_in_subdirs(path,
-#                               filename, field_dependency_name, "cloudify.*")
+# regex_to_ignore = "cloudify.*"
```

Comment:
please don't leave this in the code. you can create a manual_test.py file in the root of the module to run an example..